haskell/cabal.bzl

```
@@ -141,7 +142,13 @@ def _prepare_cabal_inputs(
     direct_libs = get_ghci_library_files(hs, cc.cc_libraries_info, cc.cc_libraries)
     transitive_libs = get_ghci_library_files(hs, cc.cc_libraries_info, cc.transitive_libraries)
     env = dict(hs.env)
-    env["PATH"] = join_path_list(hs, _binary_paths(tool_inputs) + posix.paths)
+
+    # Because python_stub_template.txt uses the shebang '/usr/bin/env python'
+    # we must have python in PATH to execute cabal_wrapper.py.tpl.
+    # It's put last to minimize disturbance.
+    # See https://github.com/tweag/rules_haskell/issues/1173
+    py_path = [paths.dirname(py.py3_runtime.interpreter_path)]
+    env["PATH"] = join_path_list(hs, _binary_paths(tool_inputs) + posix.paths + py_path)
```

Note, this assumes that a `python` binary exists in `py_path`, which wouldn't be the case if it only contains a `python3` binary. This assumption holds for, say `/usr/bin/python3` on Ubuntu if Python 2 is installed as well, it also seems to hold for `pyenv` on macOS. But, I don't know if this assumption holds in general.

---

Yes I know, but we don't have something better to do right? In the context where this extension of PATH is needed (no NixOS, no nix-shell), we're relying on the system install. I had thought of using the `dirname` of both `py.py3_runtime.interpreter_path` and `py.py2_runtime.interpreter_path` but this seemed overkill as #1173 states that `python3` should suffice (I will happily change that though, if you think it is relevant).

---

> we don't have something better to do right? 

One other option is 
```suggestion
    # Because python_stub_template.txt uses the shebang '/usr/bin/env python'
    # we must have python in PATH to execute cabal_wrapper.py.tpl.
    # It's put last to minimize disturbance.
    # See https://github.com/tweag/rules_haskell/issues/1173
    py_path = [paths.dirname(py.py3_runtime.interpreter_path)]
    env["EXTRA_PATH"] = join_path_list(hs, _binary_paths(tool_inputs) + posix.paths + py_path)
```
and then `export PATH="$EXTRA_PATH:$PATH"` which works if `python` is in the original `PATH` that Bazel configures for the sandbox. If `python` is missing there too, then we're out of luck.